pkg/gofr/http/multipart_file_bind.go

```
@@ -146,13 +163,194 @@ func (uf *formData) setFieldValue(value reflect.Value, data string) (bool, error
+	}
+
+	return false, nil
+}
+
+func dereferencePointerType(value reflect.Value) reflect.Value {
+	if value.Kind() == reflect.Ptr {
+		if value.IsNil() {
+			// Initialize the pointer to a new value if it's nil
+			value.Set(reflect.New(value.Type().Elem()))
+		}
+
+		value = value.Elem() // Dereference the pointer
+	}
+
+	return value
+}
+
+func (*formData) setInterfaceValue(value reflect.Value, data any) (bool, error) {
+	if !value.CanSet() {
+		return false, fmt.Errorf("%w: %s", errUnsupportedInterfaceType, value.Kind())
+	}
+
+	value.Set(reflect.ValueOf(data))
+
+	return true, nil
+}
+
+func (uf *formData) setSliceOrArrayValue(value reflect.Value, data string) (bool, error) {
+	elemType := value.Type().Elem()
+
+	elements := strings.Split(data, ",")
+
+	// Create a new slice/array with appropriate length and capacity
+	var newSlice reflect.Value
+	if value.Kind() == reflect.Slice {
+		newSlice = reflect.MakeSlice(value.Type(), len(elements), len(elements))
+	} else if value.Kind() == reflect.Array {
+		if len(elements) > value.Len() {
+			return false, errDataLengthExceeds
+		}
+
+		newSlice = reflect.New(value.Type()).Elem() // Create a new zero-valued array
+	} else {
+		return false, fmt.Errorf("%w: %s", errUnsupportedKind, value.Kind())
+	}
+
+	// Create a reusable element value to avoid unnecessary allocations
+	elemValue := reflect.New(elemType).Elem()
+
+	// Set the elements of the slice/array
+	for i, strVal := range elements {
+		// Update the reusable element value
+		if _, err := uf.setFieldValue(elemValue, strVal); err != nil {
+			return false, fmt.Errorf("%w %d: %w", errSettingValue, i, err)
+		}
+
+		newSlice.Index(i).Set(elemValue)
+	}
+
+	value.Set(newSlice)
+
+	return true, nil
+}
+
+func (*formData) setStructValue(value reflect.Value, data string) (bool, error) {
+	if value.Kind() != reflect.Struct {
+		return false, errNotAStruct
+	}
+
+	dataMap, err := parseStringToMap(data)
+	if err != nil {
+		return false, err
+	}
+
+	anyFieldSet := false
```

I would return an error when no field is set this way, the return signature will be a simple err

---

I’ve updated the function to return an error if no fields were set, simplifying the return signature.